src/middleware/performanceToggles.js

```diff
@@ -0,0 +1,28 @@
+import * as types from "../actions/types";
+
+/**
+ * Performance toggles (reduxState.controls.performanceToggles) represent flags
+ * for which we enable/disable certain functionality in Auspice. These flags
+ * shouldn't be depended on, i.e. Auspice should work just fine without them
+ * (but may be a little slow).
+ */
+
+
+export const performanceToggles = (_store) => (next) => (action) => {
+  let modifiedAction;
+  switch (action.type) {
+    case types.URL_QUERY_CHANGE_WITH_COMPUTED_STATE: /* fallthrough */
+    case types.CLEAN_START: {
```

Comment:
That action type can come from a narrative which changes dataset and thus we want to recalculate the performance flags.

> All other dispatches to this middleware function just fall through implicitly rather than explicitly.

Not sure I understand explicit vs implicit here - types other than the two here don't run any of the switch blocks as there's no `default` clause (perhaps this is what you mean by implicit fall-through?).

Reply:
Oops sorry, I misunderstood the code 🤦 I was thinking that fall through meant skipping the rest of the cases, please disregard.